jena-core/src/main/java/org/apache/jena/shared/impl/PrefixMappingImpl.java

```diff
@@ -187,7 +189,7 @@ public String getNsPrefixURI( String prefix )
         
     @Override
     public Map<String, String> getNsPrefixMap()
-        { return CollectionFactory.createHashedMap( prefixToURI ); }
+        { return new LinkedHashMap<>( prefixToURI ); }
```

Comment:
I think the problems is in `pmTestAbbrev`. For test `org.apache.jena.riot.system.TestPrefixMapOverPrefixMapping#prefixMap_abbrev_10 `, when this line is called, 
https://github.com/apache/jena/blob/fc01d33e6c11eed68784f820bf2dd161868d1d37/jena-arq/src/test/java/org/apache/jena/riot/system/AbstractTestPrefixMap.java#L313
either `(p0,b)` or `(p1,)` could be returned, as order in `prefixToURI` is not predictable. But the implemetation of this function
https://github.com/apache/jena/blob/fc01d33e6c11eed68784f820bf2dd161868d1d37/jena-arq/src/test/java/org/apache/jena/riot/system/AbstractTestPrefixMap.java#L326
would only return `"p0:b"`. So if line 313 returns (p1,), the assertion here would fail.
https://github.com/apache/jena/blob/fc01d33e6c11eed68784f820bf2dd161868d1d37/jena-arq/src/test/java/org/apache/jena/riot/system/AbstractTestPrefixMap.java#L328


However, I do agree that putting the `LinkedHashMap` there is not a good fix. All that does is to make nondex, the tool I was using, not to shuffle that map so that the test does not fail. 

I hope this offers some clarification.

Reply:
Thank you for the clarification. 

The test is wrong.
It even says so!

https://github.com/apache/jena/blob/fc01d33e6c11eed68784f820bf2dd161868d1d37/jena-arq/src/test/java/org/apache/jena/riot/system/AbstractTestPrefixMap.java#L325

BTW The test is about `PrefixMap` not `PrefixMapping`. They are different.

`PrefixMapping` is the original for a model API.
It is not suitable for large scale storage even with the `PrefixMappingImpl` simplified system interface. from the way it manages the reverse mapping and because, by contract, obeys XML qname rules for abbreviation.

`PrefixMap` is the result of that experience. Simpler, lower-level. 
`PrefixMapping` is stil used in the Model API for compatibility.